MoreLinq/SkipLast.cs

```diff
@@ -0,0 +1,61 @@
+﻿#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2016 Leandro F. Vieira (leandromoh). All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Bypasses a specified number of elements at the end of the sequence.
+        /// </summary>
+        /// <param name="source">The source sequence.</param>
+        /// <param name="count">The number of elements to bypass at the end of the source sequence.</param>
+        /// <returns>
+        /// An <see cref="IEnumerable{T}"/> containing the source sequence elements except for the bypassed ones at the end.
+        /// </returns>
+        public static IEnumerable<T> SkipLast<T>(this IEnumerable<T> source, int count)
+        {
+            if (source == null) throw new ArgumentNullException("source");
+
+            if (count < 1)
+                return source;
+
+            return SkipLastImpl(source, count);
```

Comment:
@leandromoh Indeed, for `SkipLast` it would make sense to also optimize for `ICollection<T>`.  For the upstream `Take`, however, the optimization involves accessing the list [via the indexer](https://github.com/dotnet/corefx/blob/3554ed27501e6a3c478a837cbbf57e495ab78949/src/System.Linq/src/System/Linq/Partition.cs#L273), so `ICollection<T>` won't do. For our purposes however, skipping the buffering is what we want, so passing on to `Take` in the `ICollection<T>` case should still be a good optimization.